### ISSUE TITLE
Cannonical way for one time setup

### ISSUE THREAD

**Issue by stealthycoin** [awslabs/chalice#238]
What is the correct way to do one time setup at the beginning of a function? For example to load a config file?

**Comment by jamesls** [comment]
Right now, it would be just adding code at the module level, but I'm assuming we'd want something more explicit.  Something like an init() function that's guaranteed to be called only once across the entire lifecycle of the lambda function right?

**Comment by ogrodnek** [comment]
I think that's right for the behavior -- should only be run once for the lifecycle and only run before accepting requests (right now the module code gets executed for running `chalice deploy` as an example).

It would also be great if the init function had access to the chalice config environments for local runs (currently init in the module does not).

Also, I'm not sure if this is out of scope or a different feature but wondering if there could be a way for the init function to return/modify resources for use in the app?

I.e. right now I have like

```
db = MyDb()
resource = MyResource()
```

and it's easy enough to access these in my handlers.  But the problem as stated is this init code gets executed outside of just running in lambda/local, and doesn't have access to chalice config..

With the init function, I'm not sure how I'd use stuff that was created in the init function cleanly or what the best practice there would be?

wondering if there could be like an `app.context` or something? that gets populated with the return of the init function? Or that the init function could modify?

thanks!
